pkg/cluster/pod.go

```diff
@@ -503,7 +505,11 @@ func (c *Cluster) getSwitchoverCandidate(master *v1.Pod) (spec.NamespacedName, e
 	} else {
 		// in asynchronous mode find running replicas
 		for _, member := range members {
-			if PostgresRole(member.Role) != Leader && PostgresRole(member.Role) != StandbyLeader && member.State == "running" {
+			if PostgresRole(member.Role) == Leader || PostgresRole(member.Role) == StandbyLeader {
+				continue
+			}
+
+			if slices.Contains([]string{"running", "streaming", "in archive recovery"}, member.State) {
```

Comment:
The next part of the code sorts candidates by smallest amount of lag. My only concern with merging this change might cause a switchover to a replica lagging behind significantly. Overall, this is still better than what we have now, though.

https://github.com/zalando/postgres-operator/blob/45489d3741b1b956c84e4779edbfd27a26970d9e/pkg/cluster/pod.go#L511-L516

Reply:
Patroni will prevent it if this is the case.